node/operator.go

```diff
@@ -72,17 +69,26 @@ func RegisterOperator(ctx context.Context, operator *Operator, transactor core.T
 
 	logger.Info("Should call churner", "shouldCallChurner", shouldCallChurner)
 
+	// Generate salt and expiry
+
+	privateKeyBytes := []byte(operator.KeyPair.PrivKey.String())
+	salt := [32]byte{}
+	copy(salt[:], crypto.Keccak256([]byte("churn"), []byte(time.Now().String()), operator.QuorumIDs[:], privateKeyBytes))
+
+	// Get the current block number
+	expiry := big.NewInt((time.Now().Add(10 * time.Minute)).Unix())
+
 	// if we should call the churner, call it
 	if shouldCallChurner {
 		churnReply, err := requestChurnApproval(ctx, operator, churnerUrl, useSecureGrpc, logger)
 		if err != nil {
 			return fmt.Errorf("failed to request churn approval: %w", err)
 		}
 
-		return transactor.RegisterOperatorWithChurn(ctx, operator.KeyPair.PubKey, operator.Socket, operator.QuorumIDs, churnReply)
+		return transactor.RegisterOperatorWithChurn(ctx, operator.KeyPair, operator.Socket, operator.QuorumIDs, operator.PrivKey, salt, expiry, churnReply)
```

Comment:
Need to update the nodeplugin path to have this private key as well: https://github.com/Layr-Labs/eigenda/blob/master/node/plugin/cmd/main.go#L118

Reply:
Good catch. Updated!